### DISCUSSION TITLE
API rules for preventing CREATE if user already created record

### DISCUSSION THREAD

**Discussion by fjtheknight** [pocketbase/pocketbase#6071] [Q&A] [answered]
Hi. I'm using v0.23.4.
I am creating a reviews platform for barbers.
I want to prevent users from submitting more than 1 review per barber.
Reviews collection has relation fields to the barbers and users collections with fields "barber" and "user".

I can't find the right syntax for this.
I tried the following but it's still allowing the same user to create a new review for the same barber:
`
@request.auth.id != "" && @request.auth.verified = true && @request.auth.id = user.id && status = "PENDING_APPROVAL"  && (@collection.reviews.barber.id ?!= barber.id ||  @collection.reviews.user.id ?!= user.id )
`

I tried the following but got this error "Invalid rule. Raw error: invalid sign operator "!".":
`@request.auth.id != "" && @request.auth.verified = true && @request.auth.id = user.id && status = "PENDING_APPROVAL"  && !(@collection.reviews.barber.id ?= barber.id &&  @collection.reviews.user.id ?= user.id )
`

Thank you for your help

**Comment by ganigeorgiev** [comment, marked as answer]
The Create API rule executes after a dry submit (_this may change in the future_), aka. when the rule execute the record would be already available in the database.

In any case, you don't have to define an API rule for this and you could register a "UNIQUE" index for the fields pair from the "Unique constraints and indexes" section right under the collection fields.
